echidna/scripts/dump_smeared_energy.py

```diff
@@ -0,0 +1,92 @@
+""" Example smearing script
+
+This script:
+  * Reads in mc spectra from hdf5
+  * Smears spectra, default is to use weighted Gaussian method, but can
+    also use specify random Gaussian method via command line
+  * Smeared spectrum is saved to the same directory with ``_#value#ly`` or
+    ``_#value#rs`` added to the file name
+
+Examples:
+  To smear hdf5 file ``example.hdf5`` using the random Gaussian method::
+
+    $ python dump_smeared.py --smear_method "random" /path/to/example.hdf5
+
+  This will create the smeared hdf5 file ``/path/to/example_200ly.hdf5``.
+
+.. note:: Valid smear methods include:
+
+  * "weight", default
+  * "random"
+"""
+
+import echidna.output.store as store
+import echidna.core.smear as smear
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-m", "--smear_method", nargs='?', const="weight",
+                        type=str, default="weight",
+                        help="specify the smearing method to use")
+    parser.add_argument("-r", "--energy_resolution", default=None, type=float,
+                        help="specify energy resolution "
+                        "e.g. 0.05 for 5 percent")
+    parser.add_argument("-l", "--light_yield", default=200., type=float,
+                        help="specify light yield"
+                        "e.g. 200 for 200 NHit/MeV")
+    parser.add_argument("-g", "--gaus", dest="gaus", action="store_true",
+                        help="Apply gaussian PDF")
+    parser.add_argument("-d", "--dest", default=None, type=str,
+                        help="specify destination directory")
+    parser.add_argument("path", type=str,
+                        help="specify path to hdf5 file")
+    parser.set_defaults(gaus=False)
+    args = parser.parse_args()
+
+    if args.dest:
+        if os.path.isdir(args.dest):
+            directory = args.dest
+        else:
+            raise ValueError("%s does not exist" % args.dest)
+    else:
+        directory = args.path[:args.path.rfind("/")+1]  # strip filename
```

Comment:
@jwaterfield As this script is geared towards the pre-smearing (and we already have a more generic dump_smeared script) if we should make the default save directory to be the original `directory` created here plus `smeared/`. So that the default behaviour if no `--dest` argument is passed is for the output hdf5 to be saved in a `smeared` directory inside the directory containing the original hdf5. What do you think?